instrumentation/opentelemetry-instrumentation-botocore/src/opentelemetry/instrumentation/botocore/extensions/dynamodb.py

```diff
@@ -0,0 +1,410 @@
+        SpanAttributes.AWS_DYNAMODB_INDEX_NAME: _REQ_INDEX_NAME,
+        SpanAttributes.AWS_DYNAMODB_LIMIT: _REQ_LIMIT,
+        SpanAttributes.AWS_DYNAMODB_PROJECTION: _REQ_PROJECTION,
+        SpanAttributes.AWS_DYNAMODB_SELECT: _REQ_SELECT,
+    }
+    response_attributes = {
+        SpanAttributes.AWS_DYNAMODB_CONSUMED_CAPACITY: _RES_CONSUMED_CAP_SINGLE,
+    }
+
+    @classmethod
+    def operation_name(cls):
+        return "Query"
+
+
+class _OpScan(_DynamoDbOperation):
+    start_attributes = {
+        SpanAttributes.AWS_DYNAMODB_TABLE_NAMES: _REQ_TABLE_NAME,
+    }
+    request_attributes = {
+        SpanAttributes.AWS_DYNAMODB_SEGMENT: ("Segment", None),
+        SpanAttributes.AWS_DYNAMODB_TOTAL_SEGMENTS: ("TotalSegments", None),
+        SpanAttributes.AWS_DYNAMODB_ATTRIBUTES_TO_GET: _REQ_ATTRS_TO_GET,
+        SpanAttributes.AWS_DYNAMODB_CONSISTENT_READ: _REQ_CONSISTENT_READ,
+        SpanAttributes.AWS_DYNAMODB_INDEX_NAME: _REQ_INDEX_NAME,
+        SpanAttributes.AWS_DYNAMODB_LIMIT: _REQ_LIMIT,
+        SpanAttributes.AWS_DYNAMODB_PROJECTION: _REQ_PROJECTION,
+        SpanAttributes.AWS_DYNAMODB_SELECT: _REQ_SELECT,
+    }
+    response_attributes = {
+        SpanAttributes.AWS_DYNAMODB_COUNT: ("Count", None),
+        SpanAttributes.AWS_DYNAMODB_SCANNED_COUNT: ("ScannedCount", None),
+        SpanAttributes.AWS_DYNAMODB_CONSUMED_CAPACITY: _RES_CONSUMED_CAP_SINGLE,
+    }
+
+    @classmethod
+    def operation_name(cls):
+        return "Scan"
+
+
+class _OpUpdateItem(_DynamoDbOperation):
+    start_attributes = {
+        SpanAttributes.AWS_DYNAMODB_TABLE_NAMES: _REQ_TABLE_NAME,
+    }
+    response_attributes = {
+        SpanAttributes.AWS_DYNAMODB_CONSUMED_CAPACITY: _RES_CONSUMED_CAP_SINGLE,
+        SpanAttributes.AWS_DYNAMODB_ITEM_COLLECTION_METRICS: _RES_ITEM_COL_METRICS,
+    }
+
+    @classmethod
+    def operation_name(cls):
+        return "UpdateItem"
+
+
+class _OpUpdateTable(_DynamoDbOperation):
+    start_attributes = {
+        SpanAttributes.AWS_DYNAMODB_TABLE_NAMES: _REQ_TABLE_NAME,
+    }
+    request_attributes = {
+        SpanAttributes.AWS_DYNAMODB_ATTRIBUTE_DEFINITIONS: (
+            "AttributeDefinitions",
+            _conv_list_to_json_list,
+        ),
+        SpanAttributes.AWS_DYNAMODB_GLOBAL_SECONDARY_INDEX_UPDATES: (
+            "GlobalSecondaryIndexUpdates",
+            _conv_list_to_json_list,
+        ),
+        SpanAttributes.AWS_DYNAMODB_PROVISIONED_READ_CAPACITY: _REQ_PROV_READ_CAP,
+        SpanAttributes.AWS_DYNAMODB_PROVISIONED_WRITE_CAPACITY: _REQ_PROV_WRITE_CAP,
+    }
+
+    @classmethod
+    def operation_name(cls):
+        return "UpdateTable"
+
+
+################################################################################
+# DynamoDB extension
+################################################################################
+
+_OPERATION_MAPPING = {
+    op.operation_name(): op
+    for op in globals().values()
```

Comment:
Just curious, why use `globals` and a series of `if`s if the list of classes can be known beforehand? Is it expected that this module will receive more classes at runtime?